espresso/devnet-tests/withdraw_test.go

```diff
@@ -6,328 +6,187 @@ import (
+		return opts
 	}
-	require.NoError(t, err)
-}
-
-// depositOnL1Bridge deposits ETH on L1 to fund withdrawals
-func depositOnL1Bridge(d *Devnet, ctx context.Context, t *testing.T, depositAmount *big.Int) {
-	rollupConfig, err := d.RollupConfig(ctx)
-	require.NoError(t, err)
 
-	depositContract, err := bindings.NewOptimismPortal(rollupConfig.DepositContractAddress, d.L1)
-	require.NoError(t, err)
-
-	l1ChainID, err := d.L1.ChainID(ctx)
-	require.NoError(t, err)
-	opts, err := bind.NewKeyedTransactorWithChainID(d.secrets.Alice, l1ChainID)
-	require.NoError(t, err)
-	opts.Value = depositAmount
-	opts.GasLimit = 1_000_000
-	gasPrice, err := d.L1.SuggestGasPrice(ctx)
-	require.NoError(t, err)
-	opts.GasPrice = gasPrice
-
-	// Deposit to dummy address
-	depositTx, err := depositContract.DepositTransaction(opts, common.Address{0xff, 0xff}, depositAmount, 21000, false, nil)
-	require.NoError(t, err)
-
-	depositReceipt, err := wait.ForReceiptOK(ctx, d.L1, depositTx.Hash())
-	require.NoError(t, err)
-	require.Equal(t, types.ReceiptStatusSuccessful, depositReceipt.Status)
-}
-
-func proveWithdrawalTransaction(d *Devnet,
-	ctx context.Context,
-	t *testing.T,
-	tx *types.Transaction,
-
-) (common.Hash, bindings.TypesWithdrawalTransaction) {
-
-	// Get contract addresses from SystemConfig
+	// Get contract addresses
 	systemConfig, _, err := d.SystemConfig(ctx)
 	require.NoError(t, err)
+	factoryAddr, _ := systemConfig.DisputeGameFactory(callOpts)
+	portalAddr, _ := systemConfig.OptimismPortal(callOpts)
+
+	factory, _ := nodebindings.NewDisputeGameFactoryCaller(factoryAddr, d.L1)
+	portal2, _ := nodepreview.NewOptimismPortal2(portalAddr, d.L1)
+	portal2Caller, _ := nodepreview.NewOptimismPortal2Caller(portalAddr, d.L1)
+	gameType, _ := portal2Caller.RespectedGameType(callOpts)
+
+	// Step 1: Wait for proposer to start (just need 1 game)
+	t.Log("Waiting for proposer to create first game...")
+	require.Eventually(t, func() bool {
+		count, _ := factory.GameCount(callOpts)
+		if count != nil && count.Cmp(common.Big0) > 0 {
+			t.Logf("Proposer started: %d games", count)
+			return true
+		}
+		return false
+	}, 3*time.Minute, 2*time.Second, "proposer didn't start")
+
+	// Step 2: Initiate withdrawal
+	t.Log("Initiating withdrawal on L2...")
+	withdrawalAmount := big.NewInt(1e18) // 1 ETH
+
+	l2MessagePasser, _ := bindings.NewL2ToL1MessagePasser(
+		common.HexToAddress("0x4200000000000000000000000000000000000016"), d.L2Seq)
+	l2ChainID, _ := d.L2Seq.ChainID(ctx)
+	l2Opts, _ := bind.NewKeyedTransactorWithChainID(d.secrets.Alice, l2ChainID)
+	l2Opts.Value = withdrawalAmount
+
+	withdrawTx, err := l2MessagePasser.InitiateWithdrawal(l2Opts, alice, big.NewInt(21000), nil)
+	require.NoError(t, err)
+	withdrawReceipt, err := wait.ForReceiptOK(ctx, d.L2Verif, withdrawTx.Hash())
+	require.NoError(t, err)
+	t.Logf("Withdrawal initiated at L2 block %d", withdrawReceipt.BlockNumber)
+
+	// Deposit ETH to L1 bridge to fund withdrawals
```

Comment:
What will happen if we skip this deposit step? Will the withdrawal fail due to insufficient balance? I'm a bit confused why we do it after the withdrawal is initialized, if the purpose is to pre-fund the balance.